pkg/detector/detector.go

```diff
@@ -1103,7 +1104,13 @@ func (d *ResourceDetector) HandlePropagationPolicyDeletion(policyID string) erro
 		return err
 	}
 
-	cleanupMarksFunc := func(obj metav1.Object) {
+	cleanupMarksFunc := func(obj metav1.Object, targetPolicyLabel string, targetPolicyID string) {
+		// When cleaning up the marks, the resource template may have already been bound to a new clusterPropagationPolicy, causing the marks' values to change.
+		// More info can refer to https://github.com/karmada-io/karmada/issues/5307.
```

Comment:
```suggestion
		// We need to ensure that the labels/annotations being deleted belong to the current PropagationPolicy,
		// otherwise, there is a risk of mistakenly deleting the ones belonging to another PropagationPolicy.
		// This situation could occur during the rapid deletion and creation of PropagationPolicy(s).
		// More info can refer to https://github.com/karmada-io/karmada/issues/5307.
```